assets/js/components/ClustersList.test.jsx

```diff
@@ -1,30 +1,127 @@
-          screen
-            .getByTestId(`filter-${filter}-options`)
-            .querySelector('li > div > span').firstChild
-        );
-      });
-
-      expect(window.location.search).toEqual(
-        '?health=unknown&name=drbd_cluster&sid=HDD&type=unknown&tags=tag1'
-      );
-    });
+  describe('filtering', () => {
+    const cleanInitialState = {
+      hostsList: {
+        hosts: [],
+      },
+      clustersList: {
+        clusters: [],
+      },
+      sapSystemsList: {
+        sapSystems: [],
+        applicationInstances: [],
+        databaseInstances: [],
+      },
+    };
+
+    const scenarios = [
+      {
+        filter: 'Health',
+        options: ['unknown', 'passing', 'warning', 'critical'],
+        state: {
+          ...cleanInitialState,
+          clustersList: {
+            clusters: [].concat(
+              clusterFactory.buildList(2, { health: 'unknown' }),
+              clusterFactory.buildList(2, { health: 'passing' }),
+              clusterFactory.buildList(2, { health: 'warning' }),
+              clusterFactory.buildList(2, { health: 'critical' })
+            ),
+          },
+        },
+        expectedRows: 2,
+      },
+      {
+        filter: 'Name',
+        options: ['cluster1', 'cluster2'],
+        state: {
+          ...cleanInitialState,
+          clustersList: {
+            clusters: [].concat(
+              clusterFactory.buildList(4),
+              clusterFactory.buildList(1, { name: 'cluster1' }),
+              clusterFactory.buildList(1, { name: 'cluster2' })
+            ),
+          },
+        },
+        expectedRows: 1,
+      },
+      {
+        filter: 'SID',
+        options: ['PRD', 'QAS'],
+        state: {
+          ...cleanInitialState,
+          clustersList: {
+            clusters: [].concat(
+              clusterFactory.buildList(4),
+              clusterFactory.buildList(2, { sid: 'PRD' }),
+              clusterFactory.buildList(2, { sid: 'QAS' })
+            ),
+          },
+        },
+        expectedRows: 2,
+      },
+      {
+        filter: 'Type',
+        options: ['hana_scale_up'],
+        state: {
+          ...cleanInitialState,
+          clustersList: {
+            clusters: [].concat(
+              clusterFactory.buildList(2, { type: 'unknown' }),
+              clusterFactory.buildList(2, { type: 'hana_scale_up' })
+            ),
+          },
+        },
+        expectedRows: 2,
+      },
+      {
+        filter: 'Tags',
+        options: ['Tag1', 'Tag2'],
+        state: {
+          ...cleanInitialState,
+          clustersList: {
+            clusters: [].concat(
+              clusterFactory.buildList(2),
+              clusterFactory.buildList(2, { tags: [{ value: 'Tag1' }] }),
+              clusterFactory.buildList(2, { tags: [{ value: 'Tag2' }] })
+            ),
+          },
+        },
+        expectedRows: 2,
+      },
+    ];
+
+    it.each(scenarios)(
+      'should filter the table content by $filter filter',
```

Comment:
What is the intent? display exactly `$filter` or interpolate it with Health/SID/Tags/etc?

Reply:
Interpolate with `Health/SID...`. This is how the interpolation is done for `it.each`

Reply:
https://jestjs.io/docs/api#1-testeachtablename-fn-timeout